cmd/controller/main.go

```diff
@@ -53,5 +71,55 @@ func main() {
 		"reconciliation-timeout", reconciler.DefaultTimeout,
 		"The amount of time to give each reconciliation of a resource to complete before its context is canceled.")
 
-	sharedmain.MainWithContext(signals.NewContext(), "controller", ctors...)
+	ctx := signals.NewContext()
+
+	// Allow configuration of threads per controller
+	if val, ok := os.LookupEnv("K_THREADS_PER_CONTROLLER"); ok {
+		threadsPerController, err := strconv.Atoi(val)
+		if err != nil {
+			log.Fatalf("failed to parse value %q of K_THREADS_PER_CONTROLLER: %v\n", val, err)
+		}
+		controller.DefaultThreadsPerController = threadsPerController
+	}
+
+	// TODO(mattmoor): Remove this once HA is stable.
+	disableHighAvailability := flag.Bool("disable-ha", false,
+		"Whether to disable high-availability functionality for this component.  This flag will be deprecated "+
+			"and removed when we have promoted this feature to stable, so do not pass it without filing an "+
+			"issue upstream!")
+
+	// HACK: This parses flags, so the above should be set once this runs.
+	cfg := injection.ParseAndGetRESTConfigOrDie()
+
+	if *disableHighAvailability {
+		ctx = sharedmain.WithHADisabled(ctx)
+	}
+
+	// If nil it panics
+	client := kubernetes.NewForConfigOrDie(cfg)
```

Comment:
Sorta don't want to copy all of this. Let's setup a kubeclient that we throw away

```go
cfg := injection.ParseAndGetRESTConfigOrDie()
kubernetes.NewForConfigOrDie(cfg)
```

Reply:
Yes but you can't then keep `sharedmain.MainWithContext` as it will call again injection.ParseAndGetRESTConfigOrDie() and flag parsing will fail with:
`panic: /ko-app/controller flag redefined: cluster`. You need to stick with `MainWithConfig` which does not have all the above logic like: `K_THREADS_PER_CONTROLLER` management etc. Do we want to remove that?

Reply:
Done, are you ok with the current update?